docs/proposals/28-autonomous-shoot-clusters.md

```diff
@@ -0,0 +1,497 @@
+5. Deploy [`gardener-resource-manager`](../concepts/resource-manager.md) in [host network mode](#host-network-vs-pod-network).
+6. Apply the corresponding provider and networking extensions in [host network mode](#host-network-vs-pod-network).
+7. Deploy `kube-proxy` and CoreDNS for service routing and domain name resolution.
+8. Create a [`Network` resource](../extensions/network.md) and apply it to generate `ManagedResource` associated with the pod network.
+9. Apply the corresponding provider and networking extensions in [pod network mode](#host-network-vs-pod-network).
+   Also deploy the operating system config extensions.
+10. Redeploy `gardener-resource-manager` in [pod network mode](#host-network-vs-pod-network).
+11. Create the [`OperatingSystemConfig` resource](../extensions/operatingsystemconfig.md) now in the cluster and activate the `gardener-node-agent` process via its `systemd` unit.
+12. Apply [`ControlPlane` resource](../extensions/controlplane.md).
+
+##### Problems & Solutions
+
+As we intend to re-use code as much as possible, we need to jump through some hoops to make this work.
+
+###### Faking a Kubernetes API
+
+Most of the Gardener code base assumes an always existing garden cluster or the API server of a seed cluster, which might not be available during bootstrapping.
+A somewhat simple solution for components running as part of the same process is to use a fake client, e.g., `sigs.k8s.io/controller-runtime/pkg/client/fake` or `github.com/gardener/gardener/pkg/client/kubernetes/fake`.
+Those fake clients, which are usually used for unit testing, can be utilized to create the necessary resources in memory without a real API server.
+It allows to simulate/fake the API server until the real one becomes available.
+
+In the `gardenadm init` flow, the initial application of the `OperatingSystemConfig` by components of the `gardener-node-agent` is an example of this.
+The `gardener-node-agent` retrieves the `OperatingSystemConfig` from a `Secret`.
+Furthermore, it also checks its local `Node` resource.
+Both are not available until after the control plane starts running.
+Therefore, a fake client is useful for bootstrapping.
+
+Another example is the deployment of the extensions.
+The existing implementation requires a garden cluster with `Controller{Registration,Deployment}` resources to apply `ControllerInstallation`s.
+However, this also works with the fake client so that extensions can be deployed during bootstrapping.
+
+###### Host Network vs. Pod Network
+
+Typically, the `gardener-resource-manager` is always deployed in the pod network.
+The same is true for the Gardener extensions.
+
+In general, it is desirable to be able to use the pod network as it offers a better separation and less potential for clashes, e.g., with regards to port reuse.
+Components most likely need to be adapted to make some ports configurable so that they do not clash, e.g., metrics and webhooks ports of extensions.
```

Comment:
I assume the gardener operator is responsible for resolving such port conflicts by configuring the used extensions properly, correct?

Reply:
Most likely, but perhaps we can already adapt our defaults (i.e., use distinct ports for the extensions) so that this doesn't become a task/head-ache for everybody. cc @ScheererJ